### ISSUE TITLE
Add pre-commit hooks for lint and formatting check

### ISSUE THREAD

**Issue by sudo-NithishKarthik** [COPS-IITBHU/hackalog-frontend#88]
**Current Situation**

We do not have a pre-commit hook set up, because of this, if someone is not using `prettier` and `eslint` as formatter and linter respectively, then, we will lose consistency in the codebase.

**Desired Situation**

We use `husky` to configure pre-commit hooks for lint and formatting check so that we ensure that the code that is being pushed is adhered to the rules that we enforce and hence consistent.

**References**

- [Husky docs](https://typicode.github.io/husky/#/)

**Comment by subodhk01** [comment]
That is true, there should be a pre-commit hook. Would you like to work on this issue?

**Comment by sudo-NithishKarthik** [comment]
Sure @subodhk01 , will put in a PR for this 👍
